gcp/main.tf

```diff
@@ -2,52 +2,111 @@
 # SPDX-License-Identifier: MPL-2.0
 
 provider "google" {
-  region  = var.google_region
-  project = var.google_project
+  region = "global"
 }
 
-# First, we create a GCP service account.
+variable "project_id" {
+  type        = string
+  description = "GCP Project ID"
+}
+
+variable "hcp_tf_organization" {
+  type        = string
+  description = "HCP Terraform Organization"
+}
 
-resource "google_service_account" "stacks_service_account" {
-  account_id   = "stacks"
+variable "hcp_tf_stacks_project" {
+  type        = string
+  description = "HCP Terraform Stacks Project"
+}
+
+# Create a service account for Terraform Stacks
+resource "google_service_account" "terraform_stacks_sa" {
+  account_id   = "terraform-stacks-sa"
   display_name = "Terraform Stacks Service Account"
+  description  = "Service account used by Terraform Stacks for GCP resources"
+}
+
+# Enable required APIs for workload identity federation
+locals {
+  gcp_service_list = [
+    "sts.googleapis.com",
+    "iam.googleapis.com",
+    "iamcredentials.googleapis.com"
+  ]
 }
 
-resource "google_service_account_key" "stacks_service_account" {
-  service_account_id = google_service_account.stacks_service_account.id
+resource "google_project_service" "services" {
+  for_each                   = toset(local.gcp_service_list)
+  project                    = var.project_id
+  service                    = each.key
+  disable_dependent_services = false
+  disable_on_destroy         = false
 }
 
-# Now, we're going to give the new service account access to things we're going to be managing in our stack.
-#
-# The policies being attached here are way to broad for a regular use case, but they make sure
-# my stacks can do anything during development and testing. In practice you should give the
-# stack access only to what it needs to manage.
+# Create Workload Identity Pool (reference google_project_service to ensure APIs are enabled)
+resource "google_iam_workload_identity_pool" "terraform_stacks_pool" {
+  depends_on = [google_project_service.services]
+  workload_identity_pool_id = "terraform-stacks-pool-3"
```

Comment:
Is there a reason this is pool 3?

Reply:
ah this was during my initial testing to prevent needing to manually remove them in gcloud console. I'll update this to remove `-3`